crates/bin/c8y_configuration_poc/src/config_manager/actor.rs

```diff
@@ -217,30 +212,47 @@ impl Actor for ConfigManagerActor {
 
 pub struct ConfigManagerMessageBox {
     pub events: mpsc::Receiver<ConfigInput>,
-    pub http_responses: mpsc::Receiver<C8YRestResult>,
-    pub http_requests: DynSender<C8YRestRequest>,
-    pub mqtt_requests: DynSender<MqttMessage>,
+    pub mqtt_publisher: DynSender<MqttMessage>,
+    pub c8y_http_proxy: C8YHttpProxy,
 }
 
 impl ConfigManagerMessageBox {
     pub fn new(
         events: mpsc::Receiver<ConfigInput>,
-        http_responses: mpsc::Receiver<C8YRestResult>,
-        http_con: DynSender<C8YRestRequest>,
-        mqtt_con: DynSender<MqttMessage>,
+        mqtt_publisher: DynSender<MqttMessage>,
+        c8y_http_proxy: C8YHttpProxy,
     ) -> ConfigManagerMessageBox {
         ConfigManagerMessageBox {
             events,
-            http_responses,
-            http_requests: http_con,
-            mqtt_requests: mqtt_con,
+            mqtt_publisher,
+            c8y_http_proxy,
         }
     }
+
+    async fn recv(&mut self) -> Option<ConfigInput> {
+        tokio::select! {
+            Some(message) = self.events.next() => {
+                match message {
+                    ConfigInput::MqttMessage(message) => {
+                        Some(ConfigInput::MqttMessage(message))
+                    },
+                    ConfigInput::FsWatchEvent(message) => {
+                        Some(ConfigInput::FsWatchEvent(message))
+                    }
+                }
+            },
+            else => None,
+        }
```

Comment:
;-) tokio::select! over a single source and a match expression that re-constructs the very same input.

```suggestion
        self.events.next().await
```